src/types/redis_set.cc

```diff
@@ -197,9 +197,14 @@ rocksdb::Status Set::MIsMember(const Slice &user_key, const std::vector<Slice> &
 }
 
 rocksdb::Status Set::Take(const Slice &user_key, std::vector<std::string> *members, int count, bool pop) {
-  int n = 0;
   members->clear();
-  if (count <= 0) return rocksdb::Status::OK();
+  if (count == 0) return rocksdb::Status::OK();
+  if (count < 0) {
+    DCHECK(!pop);
+    // NOTE: Currently, for SRANDMEMBER, we don't
+    // make duplicate members to be returned.
```

Comment:
`Currently, we don't support the returning of the same element multiple times` or something like that because `duplicate members` can be confusing since set stores only unique elements.

Reply:
Aha, let me extract a common function here